torch/quantization/make_module.py

```diff
@@ -0,0 +1,61 @@
+
+import torch
+
+_func_from_float = {
+    torch.add: torch.ops.quantized.add
+}
+
+"""Wrapper class for functions.
+
+Args:
+    func: The function to wrap the module around
+    name: Name of the module class. If None, the class name will be capitalized
+          op name.
+"""
+class _WrapModule(torch.nn.Module):
+    def __init__(self, func, name=None):
+        super(_WrapModule, self).__init__()
+        self.func = func
+        self.__name__ = self.func.__name__.capitalize() if name is None else name
+
+    def __repr__(self):
+        return self.__name__ + '()'
+
+    def forward(self, *args, **kwargs):
+        return self.func(*args, **kwargs)
+
+    def from_float(self, *args, **kwargs):
```

Comment:
Great start!. from_float is implemented as a method in quantized modules to construct it from a float module. It is required for all quantized modules so that torch.quantization.convert can have a uniform API for module conversion. . In our implementations so far, from_float is a classmethod for modules in nn.quantized. (Please see: https://github.com/pytorch/pytorch/blob/master/torch/nn/quantized/modules/linear.py for an example)

Please remove from_float() here and have it in the quantized module.
I would suggest having the following:
```PYTHON
# nn.quantized.Add
class Add(nn.Module):
      def __init__(self, out_scale=1.0, out_zero_point=0):
        super(Add, self).__init__()
        self.register_buffer('scale', torch.tensor(scale, dtype=torch.double))
        self.register_buffer('zero_point', torch.tensor(zero_point, dtype=torch.long))
     
     def forward(self, X,Y):
        return torch.ops.quantized.add(X, Y, scale =  self.scale,
                                     zero_point = self.zero_point)
        
    @staticmethod
    def from_float(mod):
        assert hasattr(mod, 'observer')
        scale, zp = mod.observer.calculate_qparams()
        return Add(scale, zp)
```

Reply:
After discussing it with @jerryzh168, I think the current approach is better. The reason is that during the convert stage, we will have to (1) create additional classes (2) create mapping from `WrapModule` to those classes, (3) Search after the mapping which op ini the wrap module corresponds to which class in `nn.quantized`.

With the current approach, both quantized and non-quantized ops can be wrapped in the same class. At the same time, we need to maintain only one dict which maps an op to its quantied counterpart.

Reply:
I agree, let us do code-gen for both quantized and original modules for functions